service/src/test/java/bio/terra/cbas/dao/TestRunSetDao.java

```diff
@@ -76,8 +76,8 @@ void retrievesSingleRunSet() {
 
   @Test
   void retrievesAllRunSets() {
-
-    List<RunSet> runSets = runSetDao.getRunSets(10);
-    assertEquals(3, runSets.size());
+    // DB has no non-templated run sets, so it will return empty list
```

Comment:
Do you want to keep the "returns everything" version of this test case too (the one that would return 3 if the `isTemplate` is true)?

Reply:
Yeah I was debating whether to keep it or not. If you think it should be present I can keep it. 

One thing to note that I had to wrap my head around is that it would be "return templated run sets" as compared to "return everything". If we want to have "return everything" for this API then the implementation of [getRunSets()](https://github.com/DataBiosphere/cbas/blob/9624648eae6b60940fc913152bcf29e3d8c01b1f/service/src/main/java/bio/terra/cbas/dao/RunSetDao.java#L27) will be different. I choose to do this way because I thought that templated run sets are side-effect of how we store inputs & outputs definiton which is a Tech Debt and might go away, and hence we might never need to return in API response. What do you think?

Reply:
+1 to not having to modifying the API. But since the switch exists in the DAO function, testing it at the DAO layer feels sensible (in case someone finds and relies on it later)